src/dbnode/persist/fs/seek_manager.go

```diff
@@ -948,30 +964,29 @@ func (m *seekerManager) openCloseLoop() {
 				byTime := m.seekersByShardIdx[elem.shard]
 				blockStartNano := xtime.ToUnixNano(elem.blockStart)
 				byTime.Lock()
-				seekers := byTime.seekers[blockStartNano]
+				seekers, ok := byTime.seekers[blockStartNano]
+				if !ok {
+					continue
+				}
+
 				allSeekersAreReturned := true
 
 				// Ensure no active seekers are still borrowed.
-				for _, seeker := range seekers.active.seekers {
-					if seeker.isBorrowed {
-						allSeekersAreReturned = false
-						break
-					}
+				if seekers.active.anyBorrowedWithLock() {
+					allSeekersAreReturned = false
 				}
 
-				// Ensure no ianctive seekers are still borrowed.
-				for _, seeker := range seekers.inactive.seekers {
-					if seeker.isBorrowed {
-						allSeekersAreReturned = false
-						break
-					}
+				// Ensure no inactive seekers are still borrowed.
+				if seekers.inactive.anyBorrowedWithLock() {
+					allSeekersAreReturned = false
+					break
```

Comment:
Not sure this should be break? It was only breaking out of the checking inactive seekers borrowed before.

Reply:
+1